staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/aggregator/downloader.go

```diff
@@ -21,32 +21,62 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"sync"
 
 	"k8s.io/apiserver/pkg/authentication/user"
 	"k8s.io/apiserver/pkg/endpoints/request"
-	"k8s.io/kube-openapi/pkg/cached"
 	"k8s.io/kube-openapi/pkg/validation/spec"
 )
 
+type CacheableDownloader interface {
+	UpdateHandler(http.Handler)
+	Get() (*spec.Swagger, string, error)
+}
+
 // cacheableDownloader is a downloader that will always return the data
 // and the etag.
 type cacheableDownloader struct {
+	name       string
 	downloader *Downloader
 	handler    http.Handler
 	etag       string
 	spec       *spec.Swagger
+
+	// mutex protects the handler which can be replaced
+	mutex sync.Mutex
```

Comment:
if the mutex protects only one variable, why don't you use an Atomic pointer instead to protect the handler?